to-do-app/src/App.js

```diff
@@ -1,26 +1,82 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
-
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+import React, { Component } from 'react';
+import Todos from './Todos'
+import AddTodo from './AddTodo'
+
+class App extends Component {
+  state = {
+    todos: [],
+    todoToShow: "all"
+  };
+
+  deleteTodo = (id) => {
+    const todos = this.state.todos.filter (todo => {
+      return todo.id !== id
+    });
+    this.setState({
+      todos
+    })
+  }
+
+ 
+
+  addTodo = (todo) => {
+    todo.id = Math.random();
+    let todos = [...this.state.todos, todo];
+    this.setState({
+      todos
+    })
+  }
+
+  toggleComplete = id => {
+    this.setState(state => ({
+      todos: state.todos.map(todo => {
+        if (todo.id === id) {
+          return {
+            ...todo,
+            complete: !todo.complete
+          };
+        } else {
+          return todo;
+        }
+      })
+    }));
+  };
+
+  updateTodoToShow = s => {
```

Comment:
```suggestion
  updateTodoToShow = showToggle => {
```
Or chose another name. Overall you should avoid single-character variables

Reply:
"s" rename to "string"